### PR TITLE
Add test for duplicate odin variable names; add error codes

### PR DESCRIPTION
**Pull request by joukevandermaas**
Error codes help when programmatically checking what went wrong.

The new test is for functionality that already worked by accident, but it supposed to work (so requires a test).

### CODE REVIEW

File: Nfield.Quota/QuotaFrameValidator.cs [hunk 1]
```diff
@@ -16,44 +16,49 @@ public QuotaFrameValidator()
...
-                .WithMessage("Quota frame definitions has variables with no levels. Affected variable definition id: '{VariableDefinitionId}'")
+                    .WithMessage("Quota frame definitions has variables with no levels. Affected variable definition id: '{VariableDefinitionId}'")
+                    .WithErrorCode("no-levels")
                 .Must(HaveValidOdinVariableName)
-                .WithMessage(
-                    "Odin variable name invalid. Odin variable names can only contain numbers, letters and '_' and cannot be empty. They can only ​start with​ a letter. First character cannot be '_' or a number. Variable definition Id '{DefId}' with name '{DefName}' has an invalid Odin Variable Name '{InvalidOdin}'");
+                    .WithMessage("Odin variable name invalid. Odin variable names can only contain numbers, letters and '_' and cannot be empty. They can only ​start with​ a letter. First character cannot be '_' or a number. Variable definition Id '{DefId}' with name '{DefName}' has an invalid Odin Variable Name '{InvalidOdin}'")
+                    .WithErrorCode("invalid-odin-variable-name");
```
Comment (mvdr) [inline Nfield.Quota/QuotaFrameValidator.cs]: It might be that .WithErrorCode() is ignored in response. 
Alternative is to use .WithState(), then it is definitely part of the response.

Comment (joukevandermaas) [reply Nfield.Quota/QuotaFrameValidator.cs]: OK let's revisit this when we've integrated in the API. Perhaps we can change the serialization there.
